snuba/admin/notifications/base.py

```diff
@@ -1,27 +1,13 @@
 from abc import ABC
 from datetime import datetime
 from enum import Enum
-from typing import Any, MutableMapping, Optional, TypedDict, Union
+from typing import Any, Mapping, Optional
 
-import structlog
 
-from snuba import settings
-from snuba.admin.notifications.slack.client import SlackClient
-from snuba.admin.notifications.slack.utils import build_blocks
-
-
-class RuntimeConfigAction(Enum):
-    ADDED = "added"
-    REMOVED = "removed"
-    UPDATED = "updated"
-
-
-ConfigType = Union[str, int, float]
-
-NotificationData = TypedDict(
-    "NotificationData",
-    {"option": str, "old": Optional[ConfigType], "new": Optional[ConfigType]},
-)
+class NotificationAction(Enum):
+    CONFIG_OPTION_ADDED = "option.added"
+    CONFIG_OPTION_REMOVED = "option.removed"
+    CONFIG_OPTION_UPDATED = "option.updated"
```

Comment:
yeah so I was thinking that maybe all the possible actions could be listed here, I dk if that makes sense though. so like for migrations I would add

```python
MIGRATIONS_RAN = "migrations.ran"
MIGRAITONS_REVERSED = "migrations.reversed"
```

or something like that.

It would be nice to be able to have one `notification_client` though and you could pass whatever action, user, timestamp, and data in. Instead of what I've kind did here where it's set up as though you'd have like a `migrations_client` as well.

Reply:
Maybe in the base class the action can be an `Any` and it's up to the child classes to implement their own version of action and maintain their own enums. Eg. `action: Any` in base, `action: RuntimeConfigAction` in child.

> It would be nice to be able to have one notification_client though and you could pass whatever action, user, timestamp, and data in.

Hmm I'm not sure, the notify function in the slack version vs the gcp log version have fairly different logic (Assuming you mean  scrapping child classes altogether)

Reply:
@rahul-kumar-saini yeah Im trying another approach in https://github.com/getsentry/snuba/pull/3406